### ISSUE TITLE
Preprocessor Check Too Early

### ISSUE THREAD

**Issue by shepting** [BranchMetrics/ios-branch-deep-linking#819]
There is a check for Minimum iOS version in several files of the Branch SDK:

```
Branch-SDK/Branch-SDK/BNCContentDiscoveryManager.m
Branch-SDK/Branch-SDK/BNCXcode7Support.h
Branch-SDK/Branch-SDK/BNCXcode7Support.m
Branch-SDK/Branch-SDK/BNCStrongMatchHelper.m
Branch-SDK/Branch-SDK/BranchCSSearchableItemAttributeSet.m
Branch-SDK/Branch-SDK/BranchCSSearchableItemAttributeSet.h
```

Unfortunately, in several of them the check happens before the import system is able to import `Foundation.h` (e.g. https://github.com/BranchMetrics/ios-branch-deep-linking/blob/master/Branch-SDK/Branch-SDK/BranchCSSearchableItemAttributeSet.h#L9).

That means that the preprocessor macros may not be defined yet. The checks just need `#import <Foundation/Foundation.h>` to occur before the 
`#if defined(__IPHONE_OS_VERSION_MAX_ALLOWED) && __IPHONE_OS_VERSION_MAX_ALLOWED >= 90000`

**Comment by E-B-Smith** [comment]
Thanks for the heads up.

I checked and found one place that the current code base was doing this and fixed it.

Here's the PR: https://github.com/BranchMetrics/ios-branch-deep-linking/pull/820  I'll merge it in with the next release which will be sometime at the end of week.

**Comment by E-B-Smith** [comment]
Fixed in release 0.24.2.
